audio_to_multiple_pose_gan/my_audio_to_pose_net/my_first_model.py

```diff
@@ -242,21 +242,33 @@ def preprocess_dataset(files):
 # compile, aka define the optimizer and loss
 model.compile(
     optimizer=tf.keras.optimizers.Adam(),
-    loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
     metrics=['accuracy'],
 )
 
+
+labs_categories = {"up": 0, "left": 1, "right": 2, "down": 3, "no": 4, "yes": 5, "go": 6, "stop": 7}
+def categorize(command: str):
+    v = np.zeros([num_labels])
+    v[labs_categories[command]] = 1.0
+    return v
+
+labs_vec = np.array([categorize(s) for s in labs])
+truncated_wavs = np.array([w[0:1000] for w in wavs])
```

Comment:
To use `model.fit` your data must be in specific formats (see tensorflow documentation). One format is as a single numpy array. However, to convert `wavs` into a numpy array (of floats, as opposed to complicated types), the length of each `wav` must be the same. I just truncated each `wav` to 1000 to get things working. Obviously you'll want to fix this. You cant just use a single numpy array though, because you want variable sized `wav` inputs. This is where my `model_runner` and `my_keras_model` allow you to deal with these details